plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidViewHolderRefactoring.java

```diff
@@ -0,0 +1,311 @@
+                        SimpleName simpleName = b.simpleName(item.variable.getIdentifier());
+                        declarationFragment.setName(simpleName);
+                        FieldDeclaration fieldDeclaration = b.getAST().newFieldDeclaration(declarationFragment);
+                        fieldDeclaration.setType(
+                                b.getAST().newSimpleType(b.simpleName(item.variable.resolveTypeBinding().getName())));
+                        viewItemsDeclarations.add(fieldDeclaration);
+                    }
+                    viewHolderItemDeclaration.modifiers().add(b.static0());
+                    r.insertBefore(viewHolderItemDeclaration, node);
+                    // create viewhHolderItem object
+                    VariableDeclarationStatement viewHolderItemVariableDeclaration = b
+                            .declareStmt(b.type("ViewHolderItem"), b.simpleName("viewHolderItem"), null);
+                    r.insertFirst(body, Block.STATEMENTS_PROPERTY, viewHolderItemVariableDeclaration);
+                    // initialize viewHolderItem
+                    Assignment viewHolderItemInitialization = b.assign(b.simpleName("viewHolderItem"),
+                            Assignment.Operator.ASSIGN, b.new0("ViewHolderItem"));
+                    statements(thenBlock).add(b.getAST().newExpressionStatement(viewHolderItemInitialization));
+                    // Assign findViewById to ViewHolderItem
+                    for (FindViewByIdVisitor.FindViewByIdItem item : findViewByIdVisitor.items) {
+                        // ensure we are accessing convertView object
+                        QualifiedName qualifiedName = b.getAST().newQualifiedName(b.name("viewHolderItem"),
+                                b.simpleName(item.variable.getIdentifier()));
+                        item.findViewByIdInvocation.setExpression(b.simpleName("convertView"));
+                        Assignment itemAssignment = b.assign(qualifiedName, Assignment.Operator.ASSIGN,
+                                (Expression) ASTNode.copySubtree(b.getAST(), item.findViewByIdExpression));
+
+                        thenBlock.statements().add(b.getAST().newExpressionStatement(itemAssignment));
+
+                        // replace previous fidnviewbyid with accesses to
+                        // viewHolderItem
+                        QualifiedName viewHolderItemFieldAccessQualifiedName = b.getAST().newQualifiedName(
+                                b.name("viewHolderItem"), b.simpleName(item.variable.getIdentifier()));
+                        r.replace(item.findViewByIdExpression, b.copy(qualifiedName));
+                    }
+                    // store viewHolderItem in convertView
+                    MethodInvocation setTagInvocation = b.invoke("convertView", "setTag",
+                            b.simpleName("viewHolderItem"));
```

Comment:
Please extract a local variable for the name of this variable `String viewHolderItemVarName` for example?